app/src/main/kotlin/com/wire/android/util/FileUtil.kt

```diff
@@ -299,19 +301,19 @@ inline fun <reified T : Parcelable> Intent.parcelableArrayList(key: String): Arr
     else -> @Suppress("DEPRECATION") getParcelableArrayListExtra(key)
 }
 
-fun Uri.getMetaDataFromUri(context: Context): FileMetaData {
-    context.contentResolver.query(this, null, null, null, null)?.use { cursor ->
-        if (cursor.moveToFirst()) {
-            val displayName =
-                cursor.getString(cursor.getColumnIndexOrThrow(MediaStore.MediaColumns.DISPLAY_NAME))
-            val size = cursor.getLong(cursor.getColumnIndexOrThrow(MediaStore.MediaColumns.SIZE))
-            return FileMetaData(displayName, size)
-        }
+@Suppress("NestedBlockDepth")
+fun Uri.getMetaDataFromUri(context: Context): FileMetaData = context.contentResolver.query(this, null, null, null, null)?.use { cursor ->
```

Comment:
Can we somehow make this function more readable? Maybe split into smaller functions?